frontend/js/src/settings/Settings.tsx

```diff
@@ -50,13 +63,79 @@ export default function Settings() {
     setShowToken(!showToken);
   };
 
+  const submitFlairPreferences = async () => {
+    if (!currentUser?.auth_token) {
+      toast.error("You must be logged in to update your preferences");
+      return;
+    }
+    const response = await APIService.submitFlairPreferences(
+      currentUser?.auth_token,
+      selectedFlairs
+    );
+    toast.success("Flair preferences updated successfully");
```

Comment:
Needs a try-catch around the API call, and handling failures as well with a toast message.